### PR TITLE
Fix bug with sgd individual learning rates

### PR DESCRIPTION
**Pull request by gpleiss**
The annealed learning rate value was calculated from the single learning rate, even if individual learning rates were supplied. This commit fixes this bug.

### CODE REVIEW

**Comment by soumith** [comment]
Looks good! Thanks!

**Comment by fmassa** [comment]
Hi, thanks for the PR!

So, the way I used to understand the `lrs` parameter was that it was relative with respect to the current `learningRate` (in which case the annealing was implemented properly).
For example, you provide `learningRate=1e-3`, and 
```lua
lrs = torch.ones(params:size())
lrs:narrow(1,1,50):fill(2)
```
I agree that the previous name (learningRates) was misleading in this context.

But this PR makes using individual `learningRates` **quite** inefficient, as [it allocates at each iteration a new tensor of the size of the parameters](https://github.com/torch/optim/pull/113/files#diff-93f332d91bf8a008fc2cb5f78ad730dfR74), which can be huge and add sync points.

**Comment by gpleiss** [comment]
@fmassa - thanks for the feedback! I was a bit confused about whether the `lrs` parameter represented absolute or relative learning rates. Regardless, this PR is more about resolving an inconsistency between `wds` and `lrs`. The `wds` parameter expected a tensor of absolute weight decays, while `lrs` expected relative learning rates.

(I do agree that relative learning rates might make more sense - in which case the `wds` parameters should be updated. That might be considered a breaking change, though?)

The additional tensor allocation was a total oversight on my part! I can fix that.

**Comment by szagoruyko** [comment]
I think we should revert this as it breaks existing code.

**Comment by fmassa** [comment]
Indeed, `wds` and `lrs` used to have different meanings, which is for sure confusing.

I have no particular objection in keeping absolute `lrs` (except from my comment in the next paragraph), but this is also a breaking change, right? (But probably not so many people use the individual learning rates due to the high memory requirements)

Yes, fixing the tensor allocation would be fine for me, but that would require having another big temporary buffer, or doing the division in place and inverting the annealing afterwards (even if it might introduce small rounding errors).

**Comment by gpleiss** [comment]
@szagoruyko - are you in favor of changing `wds` then? I realize any change made will break existing code, but I think that an inconsistency might be worse.

**Comment by szagoruyko** [comment]
@gpleiss we can't do any changes that break existing code, you can make it an option an document the behaviour better.

**Comment by soumith** [comment]
okay, looks like the immediate action item is to revert the PR. Sergey, do you have use-cases that it is breaking that Geoff can work around?

**Comment by dasguptar** [comment]
Umm, minor nitpick here, but as far as I understood, `learningRates` would need a tensor of **relative** learning rates. Going throught the source code a couple of times, I realised that `weightDecays` expects **absolute** weight decay values, not relative ones like 'learningRates'. Shouldn't this major difference between two seemingly similarly named arguments be documented somewhere?
